src/element/newElement.ts

```diff
@@ -62,17 +62,48 @@ export function newTextElement(
   return textElement;
 }
 
-export function duplicateElement(element: ReturnType<typeof newElement>) {
-  const copy = {
-    ...element,
-  };
-  if ("points" in copy) {
-    copy.points = Array.isArray(element.points)
-      ? JSON.parse(JSON.stringify(element.points))
-      : element.points;
+// Simplified deep clone for the purpose of cloning ExcalidrawElement only
+//  (doesn't clone Date, RegExp, Map, Set, Typed arrays etc.)
+//
+// Adapted from https://github.com/lukeed/klona
+function _duplicateElement(val: any, depth: number = 0) {
+  if (val == null || typeof val !== "object") {
+    return val;
+  }
+
+  let k, tmp;
+  const str = Object.prototype.toString.call(val);
+
+  if (str === "[object Object]") {
+    tmp =
+      typeof val.constructor === "function"
+        ? Object.create(Object.getPrototypeOf(val))
+        : {};
```

Comment:
Let's get rid of this, we shouldn't be copying functions that way

Reply:
This is actually to copy the prototype, and I took it from [lodash](https://github.com/lodash/lodash/blob/588bf3e20db0ae039a822a14a8fa238c5b298e65/.internal/initCloneObject.js), instead of using `__proto__` (which is slow).